src/qutip_qip/circuit/circuitsimulator.py

```diff
@@ -293,10 +310,17 @@ def __init__(
             )
 
     @property
-    def qc(self):
+    def qc(self) -> QubitCircuit:
         return self._qc
 
-    def initialize(self, state=None, cbits=None, measure_results=None):
+    def initialize(
+        self,
+        state: Optional[Qobj] = None,
+        cbits: Optional[List[int]] = None,
+        measure_results: Optional[
+            Union[List[int], Tuple[str, str], Tuple[str, str, str], Tuple[()]]
+        ] = None,
```

Comment:
This is wrong again. I guess probably we should not return tuples with undetermined length in the code and should use a list instead? What do you think?

This should be list of int or str.